clients/flash/air-client/src/org/bigbluebutton/air/screenshare/views/ScreenshareDock.as

```diff
@@ -0,0 +1,224 @@
+		}
+		
+		private function toggleStageVideo(on:Boolean):void {
+			_ns = new NetStream(_conn);
+			_ns.addEventListener(NetStatusEvent.NET_STATUS, onNetStatus);
+			_ns.addEventListener(AsyncErrorEvent.ASYNC_ERROR, onAsyncError);
+			_ns.client = this;
+			_ns.bufferTime = 0;
+			_ns.receiveVideo(true);
+			_ns.receiveAudio(false);
+			
+			//trace("**** Container w=" + this.width + ",h=" + this.height + " video w=" + _origVidWidth + ",h=" + _origVidHeight);
+			
+			var viewPort:Rectangle = positionAndSize(this.width, this.height, _origVidWidth, _origVidHeight);
+			
+			// To choose StageVideo attach the NetStream to StageVideo 
+			if (on) {				
+				_usingStageVideo = true;
+				if (_sv == null) {
+					//trace("***** Using StageVideo length=" + stage.stageVideos.length);
+					_sv = stage.stageVideos[0];
+					_sv.viewPort = viewPort;
+					_sv.addEventListener(StageVideoEvent.RENDER_STATE, stageVideoStateChange);
+					_sv.attachNetStream(_ns);
+				}
+				
+				if (_usingVideo) {
+					// If you use StageVideo, remove from the display list the 
+					// Video object to avoid covering the StageVideo object 
+					// (which is always in the background) 
+					stage.removeChild(_ssView);
+					_usingVideo = false;
+				}
+			} else {
+				// Otherwise attach it to a Video object 
+				if (_usingStageVideo) {
+					_usingStageVideo = false;
+				}
+				
+				//trace("***** Using classic Video");
+				_ssView = new ScreenshareView();
+				_ssView.x = viewPort.x;
+				_ssView.y = viewPort.y;
+				_ssView.width = viewPort.width;
+				_ssView.height = viewPort.height;
+				addElement(_ssView);
+				_ssView.display(_ns, viewPort.width, viewPort.height);
+				_usingVideo = true;
+			}
+			
+			if (!_played) {
+				_played = true;
+				//trace("***** Playing stream " + _streamId);
+				_ns.play(_streamId);
+			}
+		}
+		
+		private function stageVideoStateChange(event:StageVideoEvent):void {
+			var status:String = event.status;
+			//trace("***** stageVideoStateChange " + status + ",codec=" + event.codecInfo + ",color=" + event.colorSpace);
+		}
+			
+		public function viewStream(conn:NetConnection, streamId:String, width:int, height:int):void {
+			//trace("************ ScreenshareView: viewing of screenshare streamId=" + streamId + " w=" + width + " h=" + height);
+			_conn = conn;
+			_streamId = streamId;
+			_origVidWidth = width;
+			_origVidHeight = height;
+			
+			_screenshareRunningListener(true);
+			
+			if (stage == null) {
+				//trace("************ ScreenshareView: STAGE IS NULL!!!!!!!!");
+			} else {
+				stage.addEventListener(StageVideoAvailabilityEvent.STAGE_VIDEO_AVAILABILITY, onStageVideoState);
+			}
+		}
+		
+		public function streamStopped(session:String, reason:String):void {
+			//trace("TODO: Need to implement stopping screenshare stream");
+			_screenshareRunningListener(false);
+			stopViewing();
+		}
+		
+		override protected function updateDisplayList(w:Number, h:Number):void {
+			super.updateDisplayList(w, h);

```

Comment:
The updateDisplayList function needs to be hooked up to update the position of whichever video type is currently being played (if one is active). Currently when you rotate the screen the video has the previous position and size.